Flaky test: test_input_action_task_button[firefox] (current_time output empty after 5s)

Failed on [#2379](https://github.com/posit-dev/py-shiny/pull/2379) CI — a diff that touches only a release-skill reference script, so unrelated to the change. **Passed on rerun.**

**Job (attempt 1):** https://github.com/posit-dev/py-shiny/actions/runs/30420506896/job/90476133525 — `playwright-shiny (3.10, firefox, 1)`

## Error

```
FAILED tests/playwright/shiny/inputs/input_task_button/test_input_task_button.py::test_input_action_task_button[firefox]
  AssertionError: Locator expected not to be empty
  Actual value: empty
  Call log:
    - Expect "to_be_empty" with timeout 5000ms
    - waiting for locator("pre#current_time.shiny-text-output")
      6 × locator resolved to <pre id="current_time" aria-live="polite" class="shiny-text-output shiny-bound-output recalculating"></pre>
        - unexpected value "empty"

tests/playwright/shiny/inputs/input_task_button/test_input_task_button.py:31: AssertionError
```

The failing line is the early sanity check, before any task-button interaction:

```python
y.set("4")
result = controller.OutputCode(page, "show_result")
current_time = controller.OutputCode(page, "current_time")
# Make sure the time has content
current_time.expect.not_to_be_empty()   # line 31
```

## Observation

The output stayed `recalculating` **and** empty across all 6 locator resolutions spanning the full 5000 ms default timeout. So this is not a value-mismatch race — `current_time` simply never delivered its first value inside the window.

One thing that may be relevant: `app.py` blocks the session during startup. Both event effects are declared `ignore_none=False`, so they fire on the initial flush, and `handle_click2` awaits a 1.5 s `slow_input_compute()` which (by the app's own design and comments) blocks the session:

```python
@reactive.effect
@reactive.event(input.btn_block, ignore_none=False)
async def handle_click2():
    ...
    await slow_input_compute(input.x(), input.y())   # asyncio.sleep(1.5)
```

Meanwhile `current_time` re-invalidates every 0.1 s via `reactive.invalidate_later(0.1)`. So the assertion's 5 s budget already has a ~1.5 s startup blocking window subtracted from it, and `y.set("4")` immediately beforehand adds another flush into that same window. Under Firefox + CI load that margin may simply not be enough for the first `current_time` render to arrive.

I have **not** confirmed this mechanism — the test passed on rerun and I did not reproduce it locally. Offering it only as the most plausible starting point.

## Possible directions

- Give the line-31 assertion an explicit timeout generous enough to cover the startup blocking window, rather than relying on the 5 s default.
- Or wait for the startup blocking window to clear (e.g. assert on `show_result` first, which the test already does immediately afterwards on line 34) before asserting that `current_time` is non-empty.

## Notes

- Not a v1.7.0 regression; unrelated to the release diff.
- No pre-existing issue found for this test.
- Second distinct Firefox playwright flake observed during the v1.7.0 release train; the other is #2378 (`test_auto_scroll_pinning[firefox]`). No evidence yet that they share a cause.

--- a/tests/playwright/shiny/inputs/input_task_button/test_input_task_button.py
+++ b/tests/playwright/shiny/inputs/input_task_button/test_input_task_button.py
@@ -25,11 +25,17 @@ def test_input_action_task_button(page: Page, local_app: ShinyAppProc) -> None:
     page.goto(local_app.url)
     result = controller.OutputCode(page, "show_result")
     current_time = controller.OutputCode(page, "current_time")
+    renders_during_block = controller.OutputCode(page, "renders_during_block")
 
-    # Wait until shiny is stable
+    # Wait for the startup extended task to finish.
     result.expect_value("3", timeout=10 * 1000)
 
-    # Make sure the time has content
+    # The ignore_none=False blocking handler also runs during startup. This
+    # output is only updated after that handler completes, so use it as the
+    # explicit synchronization point before checking the ticking output.
+    renders_during_block.expect_value("0", timeout=10 * 1000)
+
+    # Make sure the time has content after startup has completed.
     current_time.expect.not_to_be_empty(timeout=10 * 1000)
 
     y = controller.InputNumeric(page, "y")
@@ -62,14 +68,6 @@ def test_input_action_task_button(page: Page, local_app: ShinyAppProc) -> None:
     # set up Blocking test
     y.set("15")
     result.expect_value("5")
-
-    # Blocking verification. The server records how many current_time renders
-    # occur inside each blocking window; this avoids sampling the display on a
-    # timer from the client, which races against in-flight output updates.
-    renders_during_block = controller.OutputCode(page, "renders_during_block")
-    # The `ignore_none=False` handler already ran one blocking window at
-    # startup; it must not have allowed any renders either.
-    renders_during_block.expect_value("0")
 
     button_block = controller.InputTaskButton(page, "btn_block")
     button_block.expect_label_busy("\n  \n Blocking...")

--- a/tests/playwright/shiny/inputs/input_task_button/test_input_task_button.py
+++ b/tests/playwright/shiny/inputs/input_task_button/test_input_task_button.py
@@ -23,15 +23,17 @@ def click_extended_task_button(
 
 def test_input_action_task_button(page: Page, local_app: ShinyAppProc) -> None:
     page.goto(local_app.url)
-    y = controller.InputNumeric(page, "y")
-    y.set("4")
     result = controller.OutputCode(page, "show_result")
     current_time = controller.OutputCode(page, "current_time")
-    # Make sure the time has content
-    current_time.expect.not_to_be_empty()
 
     # Wait until shiny is stable
-    result.expect_value("3")
+    result.expect_value("3", timeout=10 * 1000)
+
+    # Make sure the time has content
+    current_time.expect.not_to_be_empty(timeout=10 * 1000)
+
+    y = controller.InputNumeric(page, "y")
+    y.set("4")
 
     # Extended task
     button_task = controller.InputTaskButton(page, "btn_task")

